src/Neo.VM/Utility.cs

```diff
@@ -61,30 +60,24 @@ public static BigInteger Sqrt(this BigInteger value)
             return z;
         }
 
-#if !NET5_0_OR_GREATER
-        static int GetBitLength(this BigInteger i)
+        /// <summary>
+        /// Gets the number of bits required for shortest two's complement representation of the current instance without the sign bit.
+        /// </summary>
+        /// <returns>The minimum non-negative number of bits in two's complement notation without the sign bit.</returns>
+        /// <remarks>This method returns 0 if the value of current object is equal to <see cref="BigInteger.Zero"/> or <see cref="BigInteger.MinusOne"/>. For positive integers the return value is equal to the ordinary binary representation string length.</remarks>
+        public static long GetBitLength(this BigInteger num)
         {
-            byte[] b = i.ToByteArray();
-            return (b.Length - 1) * 8 + BitLen(i.Sign > 0 ? b[b.Length - 1] : 255 - b[b.Length - 1]);
+            var bytes = num.ToByteArray();
+            var size = bytes.Length;
+            if (size == 0) return 0;
+            int v = bytes[size - 1]; // 8-bit value to find the log2 of 
+            if (v == 0) return (size - 1) * 8;
+            int r; // result of log2(v) will go here
+            int shift;
+            r = (v > 0xF) ? 4 : 0; v >>= r;
+            shift = (v > 0x3) ? 2 : 0; v >>= shift; r |= shift;
+            r |= v >> 1;
+            return (size - 1) * 8 + r + 1;
```

Comment:
I think that it's almost impossible to overflow an integer

Reply:
To overflow an integer, the size of the biginteger should be close to overflow it, not realistic

Reply:
Look like different. Use old!

![image](https://github.com/neo-project/neo/assets/8141309/c8354bb1-5d4b-4df4-b677-5374ebdb6889)

Reply:
The current one doesn't take sign into account.
![image](https://github.com/neo-project/neo/assets/8141309/211541a6-ab87-49fa-ba43-c076d0162aa2)

Neg Number
![image](https://github.com/neo-project/neo/assets/8141309/38a4bbef-38e6-4cc3-9cc0-e4159c4616a1)

Reply:
We need some UTs for this method

Reply:
As in `current` I mean the one I recommended.

Reply:
yes, it doesn't work with negatives, any idea without reflection?

Reply:
In VM was not possible to use with negative numbers
https://github.com/neo-project/neo/blob/de7fc938b6460b2c1002262f4660844901035051/src/Neo.VM/Utility.cs#L49

Reply:
Copy from bouncy castle biginteger? https://www.bouncycastle.org/csharp/archive/BigInteger.cs

Reply:
```csharp
private static long Log2BitLength(BigInteger integer)
{
    return (long)Math.Ceiling(BigInteger.Log(integer.Sign < 0 ? -integer : integer + 1, 2.0));
}

private static bool TryIterativeBitLength(BigInteger integer, out long bitLength)
{
    long value;

    try
    {
        value = (long) integer;
    }
    catch (OverflowException)
    {
        bitLength = 0;
        return false;
    }

    const long signMask = unchecked((long)0x8000_0000_0000_0000);

    long signBit = value < 0 ? signMask : 0;
    long tmp = value;
    long j;
    for (j = 0; j < 64 && (tmp & signMask) == signBit; j++)
        tmp <<= 1;

    bitLength = 64 - j;
    return true;
}

private static void VerifyGetBitLength(BigInteger bigInt)
{
    long actualBitLength = GetBitLength(bigInt);
    long expectedBitLength = Log2BitLength(bigInt);

    // self-consistency check
    if (TryIterativeBitLength(bigInt, out long expectedBitLengthIterative))
        Assert.Equal(expectedBitLength, expectedBitLengthIterative);

    Assert.Equal(expectedBitLength, actualBitLength);
}

private static long GetBitLength(BigInteger bigInt)
{
    long actualBitLength = bigInt.GetBitLength();
    Assert.InRange(actualBitLength, 0, long.MaxValue);
    return actualBitLength;
}
```

Reply:
TryIterativeBitLength is for small numbers, we can try with Log2BitLength

Reply:
Im working on something give a few

Reply:
Look at `VerifyGetBitLength`

Reply:
Nice! it works!